File: backends/fiber/internal/models/recipe.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Recipe struct {
	ID          uuid.UUID    `gorm:"column:recipe_id;primaryKey;default:gen_random_uuid()" json:"id"`
	Title       string       `gorm:"column:title;not null" json:"title"`
	Description string       `gorm:"column:description" json:"description"`
	Visibility  string       `gorm:"column:visibility;default:public" json:"visibility"`
	UserID      uuid.UUID    `gorm:"column:user_id" json:"-"`
	User        User         `json:"user,omitempty"`
	Ingredients []Ingredient `gorm:"foreignKey:RecipeID" json:"ingredients,omitempty"`
	Steps       []Step       `gorm:"foreignKey:RecipeID" json:"steps,omitempty"`
	Photos      []Photo      `gorm:"foreignKey:RecipeID" json:"photos,omitempty"`
	Comments    []Comment    `gorm:"foreignKey:RecipeID" json:"comments,omitempty"`
	CreatedAt   time.Time    `gorm:"column:created_at;default:CURRENT_TIMESTAMP" json:"created_at"`
	UpdatedAt   time.Time    `gorm:"column:updated_at;default:CURRENT_TIMESTAMP" json:"updated_at"`
}

File: backends/aspnet/Recipes.API/entities/Recipe.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace Recipes.API.entities;

public partial class Recipe
{
    [JsonPropertyName("id")]
    public Guid RecipeId { get; set; }

    public string Title { get; set; } = null!;

    public string? Description { get; set; }

    public string? Visibility { get; set; }

    public Guid? UserId { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public virtual ICollection<Comment> Comments { get; set; } = new List<Comment>();

    public virtual ICollection<Ingredient> Ingredients { get; set; } = new List<Ingredient>();

    public virtual ICollection<Photo> Photos { get; set; } = new List<Photo>();

    public virtual ICollection<Rating> Ratings { get; set; } = new List<Rating>();

    public virtual ICollection<Step> Steps { get; set; } = new List<Step>();

    public virtual User? User { get; set; }
}
